### PR TITLE
[code] github auth provider

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

This PR hooks our generic auth provider as github auth provider in VS Code. The default github authentication extension is excluded. `github.vscode-pull-request-github` is installed if a workspace is started from github.com context. You can actually review this PR using this PR from VS Code: https://ak-github-auth.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod/pull/3349

Additionally it fixes bugs in the git token provider of supervisor which before will return any token regardless of given scopes.

Changes in code: https://github.com/gitpod-io/vscode/compare/gp-code...ak/generic_auth

#### How to test

- Switch to Code
- Start a GitHub workspace
- Check that GitHub pull request review extension is installed but not signed in
- Check that pushing from the cli is not allowed, not enough permissions
- Go to Access Control and grant access to public and private repos
- Sign in the pull request review extension or reload the page, check that it works now
- Check that pushing from the cli now works
- Reload the page check again
- Restart the workspace check again
- Start another workspace and check again, like on this PR to check out PR review mode: https://ak-github-auth.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod/pull/3349

### CODE REVIEW

File: components/server/src/github/scopes.ts [hunk 1]
```diff
@@ -7,18 +7,19 @@
 
 export namespace GitHubScope {
     export const EMAIL = "user:email";
+    export const READ_USER = "read:user";
```
Comment (akosyakov) [inline components/server/src/github/scopes.ts:10]: test

**Comment by akosyakov** [comment]
There is a known issue that after first sign-in the notification does not go await from the account menu. It seems to be a bug in VS Code itself, I will try to handle after next upgrade (next week).

**Review by JanKoehnlein — APPROVED** [review]
LGTM

**Comment by JanKoehnlein** [comment]
If you want to be more clever than me while testing this, create a new branch before trying to push changes. Otherwise werft will delete your deployment and kill your running test WS.

**Review by csweichel — APPROVED** [review]
Code LGTM

**Pull request merged**